pyflakes/checker.py

```diff
@@ -314,6 +326,28 @@ def __init__(self, tree, filename='(none)', builtins=None,
         self.popScope()
         self.checkDeadScopes()
 
+    @property
+    def futuresAllowed(self):
```

Comment:
this can be made more efficient, and simpler.

also need to re-add docstrings, and comments.